build/common.sh

```diff
@@ -203,6 +207,18 @@ function kube::build::is_osx() {
   [[ "$(uname)" == "Darwin" ]]
 }
 
+function kube::build::update_dockerfile() {
+  if kube::build::is_osx; then
```

Comment:
@ixdy It works, I tested on both OSes. 
I wrote simple scripts on my local laptop and verify it before I made the change, and the build succeeded on both OSes. 
Checking my local build output, the  
/Users/fmeng/code/kubernetes/src/k8s.io/validate/kubernetes/_output/images/kube-build:build-b4a36a2a04/Dockerfile
I can see the proxy settings are there.
LM-SHC-16501199:kubernetes fmeng$ more _output/images/kube-build\:build-b4a36a2a04/Dockerfile
Here is the sample without environment variable set.
FROM kube-build:cross-1.5.3-1

MAINTAINER  Joe Beda <jbeda@google.com>

# (set an explicit GOARM of 5 for maximum compatibility)
ENV GOARM 5
ENV GOOS    linux
ENV GOARCH  amd64
ENV http_proxy ""
ENV https_proxy ""
ENV no_proxy 127.0.0.1